testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -753,53 +753,93 @@ def test_persistent_offsets_size_mismatch(self):
-                     ('Reload offsets from trajectory\n ctime or size did not match', ))
-
-    # TODO: This doesn't test if the offsets work AT ALL. the old
-    # implementation only checked if the offsets were ok to set back to the old
-    # frame. But that doesn't check if any of the other offsets is potentially
-    # wrong. Basically the only way to check that would be to scan through the
-    # whole trajectory.
-    # @dec.slow
-    # def test_persistent_offsets_last_frame_wrong(self):
-    #     # check that stored offsets are not loaded when the offsets
-    #     # themselves appear to be wrong
-    #     with open(XDR.offsets_filename(self.traj), 'rb') as f:
-    #         saved_offsets = {k: v for k, v in six.iteritems(np.load(f))}
-    #     saved_offsets['offsets'] += 1
-    #     with open(XDR.offsets_filename(self.traj), 'wb') as f:
-    #         np.savez(f, **saved_offsets)
-
-    #     # with warnings.catch_warnings():
-    #     #     u = MDAnalysis.Universe(self.top, self.traj)
-    #     #     assert_equal((u.trajectory._xdr.offsets is None), True)
+                     ('Reload offsets from trajectory\n ctime, size or natoms did not match', ))
+
+    @dec.slow
+    def test_persistent_offsets_one_frame_wrong(self):
+        fname = XDR.offsets_filename(self.traj)
+        saved_offsets = XDR.read_numpy_offsets(fname)
+
+        # offset from the 4th frame is now wrong
+        idx_frame = 3
+        saved_offsets['offsets'][idx_frame] += 4
+        np.savez(fname, **saved_offsets)
+
+        with warnings.catch_warnings(record=True) as warn:
+            warnings.simplefilter('always')
+            reader = self._reader(self.traj)
+            reader[idx_frame]
+
+        assert_equal(warn[0].message.args,
+                     ("seeking didn't work, Recalculating offsets and trying again.", ))
+
+    def _save_corrupt_trj(self, fname):
+        xdr = xdrlib.Packer()
+        n_atoms = 5
+        offsets = [0]
+        offsets.append(self.pyxdr.pack_frame(xdr, 1, n_atoms=n_atoms))
+        offsets.append(self.pyxdr.pack_frame(xdr, 2, n_atoms=n_atoms))
+        offsets.append(self.pyxdr.TRR.pack_frame(xdr, 3, n_atoms=n_atoms,
+                                                 magic=42))
+        offsets.append(pylibxdr.TRR.pack_frame(xdr, 4, n_atoms=n_atoms))
+        pylibxdr.TRR.pack_frame(xdr, 4)
+
+        fname = self.tmpdir.name + fname + '.xdr'
+        with open(fname, 'wb') as f:
+            f.write(xdr.get_buffer())
+        ctime = os.path.getctime(fname)
+        size = os.path.getsize(fname)
+        offsets_filename = XDR.offsets_filename(fname)
+        np.savez(offsets_filename,
+                 offsets=offsets, size=size, ctime=ctime, n_atoms=n_atoms)
+
+        return fname
+
+    @raises(IOError)
+    @dec.slow
+    def test_throw_after_offset_recalc(self):
+        # create temp dir
+        fname = 'raises_with_corupt_file'
+        fname = self.save_corrupt_trj(fname)
```

Comment:
This creates a corrupted xtc file and checkes if the user get an exception after we tried to recalculate the offsets.